api/lib/mno_enterprise/concerns/controllers/jpi/v1/marketplace_controller.rb

```diff
@@ -82,14 +82,14 @@ def parent_organization_id
   def fetch_apps
     Rails.cache.fetch("marketplace/index-apps-#{@last_modified}-#{I18n.locale}-#{parent_organization_id}") do
       apps = MnoEnterprise::App.fetch_all(app_relation(parent_organization_id).includes(:app_shared_entities, { app_shared_entities: :shared_entity }).where(active: true))
-      apps.sort_by! { |app| [app.rank ? 0 : 1, app.rank] } # the nil ranks will appear at the end
+      apps.sort_by! { |app| [app.rank ? 0 : 1, app.rank, app.name ? app.name.downcase : ''] } # the nil ranks will appear at the end. Nil names at the start
```

Comment:
If the apps are already sorted alphabetically, sort_by will just re-arrange them by rank, so we can do:
```ruby
apps = MnoEnterprise::App.fetch_all(app_relation(parent_organization_id).includes(:app_shared_entities, { app_shared_entities: :shared_entity }).where(active: true)).order(:name)
apps.sort_by! { |app| [app.rank ? 0 : 1, app.rank] } # the nil ranks will appear at the end
```